src/item.cpp

```diff
@@ -2486,24 +2487,42 @@ int item::precise_unit_volume() const
     return volume() * 1000;
 }
 
-int item::volume( bool integral ) const
+int corpse_volume( m_size corpse_size )
+{
+    switch( corpse_size ) {
+        case MS_TINY:    return    3;
+        case MS_SMALL:   return  120;
+        case MS_MEDIUM:  return  250;
+        case MS_LARGE:   return  370;
+        case MS_HUGE:    return 3500;
+    }
+    debugmsg( "unknown monster size for corpse" );
+    return 0;
+}
+
+int item::base_volume() const
```

Comment:
Alternatively if the fire code is the only consumer just move this logic within the fire code and add suitable documentation